compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -921,12 +947,16 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 AssocItemKind::Fn(box Fn { sig, .. }) => {
                     hir::AssocItemKind::Fn { has_self: sig.decl.has_self() }
                 }
+                AssocItemKind::Delegation(..) => {
+                    hir::AssocItemKind::Fn { has_self: self.delegation_has_self(i.id, i.span) }
+                }
                 AssocItemKind::MacCall(..) => unimplemented!(),
             },
             trait_item_def_id: self
                 .resolver
                 .get_partial_res(i.id)
-                .map(|r| r.expect_full_res().def_id()),
```

Comment:
Why this change?
We should never put partial resolution into the table for these IDs due to `defer_to_typeck == false` for `PathSource::Delegation`.